### PR TITLE
Merging matlab files from matlab4compile

### PR DESCRIPTION
**Pull request by bendhouseart**
- copied over changed matab files from matlab4compile
- did rough merge w/ current state of main

Submitting as PR to further inspect.

### CODE REVIEW

File: matlab/updatejsonpetfile.m [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 % generic function that updates PET json file with missing PET-BIDS
 % information, if only the jsonfile is provided, it only checks if valid
-% (and possibly updates some fields from scalar to array)
+% (and possibly updates some fields from scaler to array)
```
Comment (bendhouseart) [inline matlab/updatejsonpetfile.m:5]:
```suggestion
% (and possibly updates some fields from scalar to array)
```

**Comment by bendhouseart** [comment]
Alright @CPernet and @DrCyPhi did my best to incorporate changes from [this pr](https://github.com/openneuropet/PET2BIDS/pull/182) into something a bit more manageable. 

Take a look at and make sure to review. We'll want to tag this before it get's merged so approve w/ out merging please.

**Comment by bendhouseart** [comment]
Ugh, it broke the phantoms CI....

**Comment by CPernet** [comment]
this branch MUST not be merge to main

**Pull request closed without merging**